aws/resource_aws_cognito_identity_provider.go

```diff
@@ -0,0 +1,180 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsCognitoIdentityProvider() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsCognitoIdentityProviderCreate,
+		Read:   resourceAwsCognitoIdentityProviderRead,
+		Update: resourceAwsCognitoIdentityProviderUpdate,
+		Delete: resourceAwsCognitoIdentityProviderDelete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Delete: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"attribute_mapping": {
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+
+			"idp_identifiers": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+
+			"provider_details": {
+				Type:     schema.TypeMap,
+				Optional: true,
+				Computed: true,
+			},
+
+			"provider_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"provider_type": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"user_pool_id": {
+				Type:     schema.TypeString,
+				Required: true,
```

Comment:
Should user_pool_id be use ForceNew? If the pool id changes, wouldn't a new provider need to be created?

Reply:
👍